### ISSUE TITLE
Installation of spotify fails

### ISSUE THREAD

**Issue by c0deaddict** [NixOS/nixpkgs#41389]
## Issue description

Installation of spotify fails. 

```
installing 'spotify-1.0.79.223.g92622cc2-21'
these derivations will be built:
  /nix/store/i7bh7sf6syhijl0ii4jzq7xcydqzjhxp-spotify-client_1.0.79.223.g92622cc2-21_amd64.deb.drv
  /nix/store/xn2rwzm6rgy56165zy3d6bg09z8jk57b-spotify-1.0.79.223.g92622cc2-21.drv
building '/nix/store/i7bh7sf6syhijl0ii4jzq7xcydqzjhxp-spotify-client_1.0.79.223.g92622cc2-21_amd64.deb.drv'...

trying https://repository-origin.spotify.com/pool/non-free/s/spotify-client/spotify-client_1.0.79.223.g92622cc2-21_amd64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404
error: cannot download spotify-client_1.0.79.223.g92622cc2-21_amd64.deb from any mirror
builder for '/nix/store/i7bh7sf6syhijl0ii4jzq7xcydqzjhxp-spotify-client_1.0.79.223.g92622cc2-21_amd64.deb.drv' failed with exit code 1
cannot build derivation '/nix/store/xn2rwzm6rgy56165zy3d6bg09z8jk57b-spotify-1.0.79.223.g92622cc2-21.drv': 1 dependencies couldn't be built
error: build of '/nix/store/xn2rwzm6rgy56165zy3d6bg09z8jk57b-spotify-1.0.79.223.g92622cc2-21.drv' failed
```

### Steps to reproduce

```sh
nix-env -iA nixos.spotify
```

## Technical details

```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.40, NixOS, 18.03.132536.fdb5ba4cdf9 (Impala)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.0.2`
 - channels(root): `"nixos-18.03"`
 - channels(jos): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`
```

**Comment by joachifm** [comment]
cc @srhb

**Comment by joachifm** [comment]
I think this will be fixed by https://github.com/NixOS/nixpkgs/pull/41356

**Comment by xeji** [comment]
fixed in #41356
